packages/kbn-apm-synthtrace/src/scenarios/mobile.ts

```diff
@@ -354,34 +364,40 @@ const scenario: Scenario<ApmFields> = async ({ scenarioOpts, logger }) => {
           device.startNewSession();
           const framework =
             device.fields['device.manufacturer'] === 'Apple' ? 'iOS' : 'Android Activity';
+          const couldCrash = modelIdentifiersWithCrashes.includes(
+            device.fields['device.model.identifier'] ?? ''
+          );
+          const startTx = device
+            .transaction('Start View - View Appearing', framework)
+            .timestamp(timestamp)
+            .duration(500)
+            .success()
+            .children(
+              device
+                .span({
+                  spanName: 'onCreate',
+                  spanType: 'app',
+                  spanSubtype: 'external',
+                  'service.target.type': 'http',
+                  'span.destination.service.resource': 'external',
+                })
+                .duration(50)
+                .success()
+                .timestamp(timestamp + 20),
+              device
+                .httpSpan({
+                  spanName: 'GET backend:1234',
+                  httpMethod: 'GET',
+                  httpUrl: 'https://backend:1234/api/start',
+                })
+                .duration(800)
+                .failure()
+                .timestamp(timestamp + 400)
+            );
           return [
-            device
-              .transaction('Start View - View Appearing', framework)
-              .timestamp(timestamp)
-              .duration(500)
-              .success()
-              .children(
-                device
-                  .span({
-                    spanName: 'onCreate',
-                    spanType: 'app',
-                    spanSubtype: 'external',
-                    'service.target.type': 'http',
-                    'span.destination.service.resource': 'external',
-                  })
-                  .duration(50)
-                  .success()
-                  .timestamp(timestamp + 20),
-                device
-                  .httpSpan({
-                    spanName: 'GET backend:1234',
-                    httpMethod: 'GET',
-                    httpUrl: 'https://backend:1234/api/start',
-                  })
-                  .duration(800)
-                  .failure()
-                  .timestamp(timestamp + 400)
-              ),
+            couldCrash && index % 2 === 0
+              ? startTx.errors(device.crash({ message: 'error' }).timestamp(timestamp))
+              : startTx,
```

Comment:
You can also inline this:

```ts
transaction.errors(
	...(couldCrash && index % 2 === 0 ? [ device.crash(...)] : [])
)
```

Reply:
Created [issue](https://github.com/elastic/kibana/issues/155594) to address this

Reply:
FWIW, this doesn't need a refactor of Synthtrace, this is already possible today.